fix: disable sending a second interrupt signal to `terraform`

levkoburburas opened the pull request:
<!-- Prepend '[WIP]' to the title if this PR is still a work-in-progress. Remove it when it is ready for review! -->

## Description

When `CTRL+c` keys are pressed, `terragrunt` forwards this interrupt signal to the child process `terraform`, while the shell already passes interrupt signal to the main process (in our case `terragrunt`) and its children as well (in our case `terraform`). As a result, `terraform` receives the interrupt signal twice and cancels graceful shutdown.

```
Interrupt received.
Please wait for Terraform to exit or data loss may occur.
Gracefully shutting down...


Two interrupts received. Exiting immediately. Note that data loss may have
occurred.
```

Fixes #2120.

<!-- Description of the changes introduced by this PR. -->

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [ ] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [ ] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

<!-- One-line description of the PR that can be included in the final release notes. -->
fix: disable sending a second interrupt signal to `terraform`

### Migration Guide

<!-- Important: If you made any backward incompatible changes, then you must write a migration guide! -->

Etiene commented:
Thanks for the PR! How have you tested this? Will it not make [this issue](https://github.com/gruntwork-io/terragrunt/issues/61) resurface?

levkoburburas commented:
> How have you tested this? Will it not make [this issue](https://github.com/gruntwork-io/terragrunt/issues/61) resurface?

Yes, it could. This [comment](https://github.com/gruntwork-io/terragrunt/issues/61#issuecomment-264428096) made it clear to me that the signal can be sent in a variety of ways. I tested in a shell that sends a signal to a process group, in which case we shouldn't forward the signal, but it can also only be sent to the main process and than we need to forward the signal. Thus, we must consider all cases.

Fixed and added unit test.

Thank you for the help @Etiene!

zackproser reviewed (approved):
LGTM

levkoburburas commented:
Thank you for the review @zackproser, @denis256!
